### DISCUSSION TITLE
如何实现医疗化验单的化验项目和化验结果以及参考范围的提取

### DISCUSSION THREAD

**Discussion by Flamingo-Lin** [PaddlePaddle/PaddleOCR#13881] [Q&A]
我这边想实现一个医疗化验单的化验信息的提取，如下图所示：
![image](https://github.com/user-attachments/assets/555c07de-1265-4b8a-95bc-f27b0ff2996d)
将化验单中的项目、项目结果和参考范围提取然后转换为一个excel表、或者json等结构化的格式。例如下图
![image](https://github.com/user-attachments/assets/ece23a33-3b6d-46d2-a02d-8a821e7ea567)
（此提取的结果参考范围还没有提取出来）
如果是使用PaddleOCR的PP-Structure 文档分析的话，请问需要使用的功能有哪些？版面分析、表格识别、关键信息提取这三个功能都需要用到吗，还是？
顺便我想请问关键信息提取的话只能是一个question对应一个answer吗？可以question对应多个answer吗?(如果是化验项目为question的话，可以将化验项目结果和参考范围作为answer吗？)
感谢各位大佬的解答！

**Comment by Flamingo-Lin** [comment]
我现在的想法是使用霍夫变化检测三线表然后将图片裁剪为只有项目表的图片然后使用paddleOCR的通用表格识别模块进行后续的工作，请问大佬们这样的项目可行吗？

**Reply by Flamingo-Lin** [reply]
不过对于体检报告单可能一单中有很多的表格，如下图所示：
![image](https://github.com/user-attachments/assets/5d0a5e9a-b5b4-4467-aa7e-b28868a7deb1)
有没有什么办法可以将这里面的含有多个表格的图片裁剪为每个只含一个表格的图片呢？又或者是有什么办法可以直接识别多个表格的办法？

**Comment by UserWangZz** [comment]
可以先使用表格检测，截取出这些表格，然后使用表格识别来将图片转为excel文件，后续可以对识别结果进行进一步处理。
关于关键信息提取，理论上支持单个key关联多个value。

**Reply by Flamingo-Lin** [reply]
好的，非常感谢您的建议，我去试试看！

**Reply by Flamingo-Lin** [reply]
请问如果是自定义数据进行训练的话，数据的标注和处理有官方的格式和标准嘛？

**Reply by Flamingo-Lin** [reply]
我看到官方文档有一个PPOCRLabelv2的工具是可以使用的对嘛？

**Reply by UserWangZz** [reply]
是的，可以直接使用PPOCRLabel进行标注。
表格检测目前PaddleOCR提供了一个单独检测表格的版面分析模型，可以先尝试效果，如果可以就不需要微调模型了
目前KIE任务可以试试小模型加大模型尝试，线上提供PPChatOCR体验

**Reply by Flamingo-Lin** [reply]
我这边试了一下，版面分析的效果不是非常准确，请问版面分析这个需要微调的话，也需要对自己的数据集进行标注为paddleocr可以处理的格式嘛？这个标注是人工标注还是有可以使用的工具呢？

**Reply by UserWangZz** [reply]
版面分析吗，模型有一个单独检测表格的模型，可以试试。
版面分析base的是PaddleDetection，数据集好像是COCO格式的即可。也有相应的文档
https://paddlepaddle.github.io/PaddleOCR/ppstructure/models_list.html#1 参考该连接有具体的模型下载
